### PR TITLE
Correlation plot with nans

### PR DESCRIPTION
**Pull request by PaulJonasJost**
Added a test. There was previously an error, when "all" start indices were selected, where no general checks were run.

### CODE REVIEW

**Review by stephanmg — APPROVED** [review]
LGTM.

File: pypesto/visualize/misc.py [hunk 1]
```diff
@@ -359,6 +359,13 @@ def process_start_indices(
         if start_index < len(result.optimize_result)
     ]
 
+    # filter out the indices that are not finite
+    start_indices = [
+        start_index
+        for start_index in start_indices
+        if np.isfinite(result.optimize_result[start_index].fval)
+    ]
```
Comment (Doresic) [inline pypesto/visualize/misc.py:367]: Should we check for `np.isnan` here as well?

**Review by Doresic — COMMENTED** [review]
The `process_start_indices` function is used by multiple visualization routines. And some of them also have specific ways of dealing with infinite values. I think those should then be updated as well accordingly if `process_start_indices` doesn't return start indices with infinite values anymore. At least removed, so that the start indices are not checked for infinite values.

We should also think about adding a warning or logging entry that some start indices were removed from plotting due to infinite values (if that's the case). That's, for instance, done in the `parameters.py` plotting routine.

File: pypesto/visualize/misc.py (hunk 1 above)
Comment (stephanmg) [reply pypesto/visualize/misc.py:367]: np.isfinite tests for `Not a Number` already.

**Comment by stephanmg** [comment]
> We should also think about adding a warning or logging entry that some start indices were removed from plotting due to infinite values (if that's the case). That's, for instance, done in the `parameters.py` plotting routine.

Could that be simply a string message with: "Removed XY entries due to infinite or NaN value"?

**Review by m-philipps — APPROVED** [review]
Looks good to me, I would update the `process_start_indices` description, e.g.:
```
Create an array of indices if a number was provided, checks that the
indices do not exceed the max_index and removes starts with non-finite fval.
```

**Pull request merged**